configstore/backends/base.py

```diff
@@ -0,0 +1,10 @@
+from typing import Optional
+
+
+class Backend(object):
+
+    def __init__(self) -> None:
+        pass
+
+    def get_setting(self, key: str) -> Optional[str]:
+        raise NotImplementedError
```

Comment:
I would like this to be a real abstract base class, and also keep the duck typing (not require that classes inherit from the base class).  We could chat about this on Monday, the background and reasoning could be quite long for a review message!

(Duck typing / virtual subclasses can be handled by typing: https://mypy.readthedocs.io/en/stable/protocols.html#protocol-types )